src/modules/producer.js

```diff
@@ -13,33 +28,43 @@ function createRpcQueue(queue) {
   if (!amqpRPCQueues[queue]) {
     amqpRPCQueues[queue] = {};
   }
-
-  let rpcQueue = amqpRPCQueues[queue];
+  
+  const rpcQueue = amqpRPCQueues[queue];
   if (rpcQueue.queue) return Promise.resolve(rpcQueue.queue);
-
+  
   //we create the callback queue using base queue name + appending config hostname and :res for clarity
   //ie. if hostname is gateway-http and queue is service-oauth, response queue will be service-oauth:gateway-http:res
   //it is important to have different hostname or no hostname on each module sending message or there will be conflicts
   var resQueue = queue + ':' + this.conn.config.hostname + ':res';
-  rpcQueue.queue = this.conn.get().then((channel) => {
-    return channel.assertQueue(resQueue, { durable: true, exclusive: true })
-      .then((_queue) => {
-        rpcQueue.queue = _queue.queue;
-
-        //if channel is closed, we want to make sure we cleanup the queue so future calls will recreate it
-        this.conn.addListener('close', () => { delete rpcQueue.queue; createRpcQueue.call(this, queue); });
-
-        return channel.consume(_queue.queue, maybeAnswer.call(this, queue), { noAck: true });
-      })
-      .then(() => rpcQueue.queue);
+  rpcQueue.queue = this
+    .conn
+    .get()
+    .then((channel) => {
+      return checkQueue(channel, resQueue)
+        .then(()=> {
+          return channel.assertQueue(resQueue, {durable: true, exclusive: true})
+            .then((_queue) => {
+              rpcQueue.queue = _queue.queue;
+              
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.dialonce.io/coding_rules#rule_key=javascript%3ATrailingWhitespace)